### PR TITLE
Boyloe/add response to ls

### LINKED ISSUE #2: Write responses to local storage
On every answer selection, write updated data to local storage

### PR DESCRIPTION
**Pull request by boyloe**
I created a new src/hooks folder to store any custom hooks we might use in the project. I created a useLocalStorage hook, sourced from [useHooks](https://usehooks.com/useLocalStorage/).  Added onValueChange listener to Survey Model that updates local storage with current survey data each time a new value is selected by the user.

### CODE REVIEW

File: src/Components/SurveyComponent.js [hunk 1]
```diff
@@ -2,15 +2,23 @@ import React from 'react';
 import * as Survey from 'survey-react';
 import SURVEY_DATA from '../data/survey.js';
 import { useState, useEffect, useRef } from 'react';
+import useLocalStorage from '../hooks/useLocalStorage'
 
 function SurveyComponent() {
+
+    const [surveyData, setSurveyData] = useLocalStorage(Object.keys, Object.values)
```
Comment (eemshi) [inline src/Components/SurveyComponent.js]: It looks like the hook expects the name of the local storage key that we want to write to (any string), and its initial value, so maybe we could pass something like `useLocalStorage("surveyData", null)`. Hope that makes sense, but let me know if I'm missing something

File: src/Components/SurveyComponent.js [hunk 2]
```diff
@@ -2,15 +2,23 @@ import React from 'react';
...
+    }
+
+    function persistDataToLocalStorage(survey)  {
+        setSurveyData(survey.data)
     }
 
     useEffect(() => {});
 
-    return <Survey.Survey json={SURVEY_DATA} onComplete={sendDataToServer} />;
+    return <Survey.Survey json={SURVEY_DATA}   onValueChanged={persistDataToLocalStorage} onComplete={sendDataToServer} />;
```
Comment (eemshi) [inline src/Components/SurveyComponent.js:21]: 🙌

File: src/Components/SurveyComponent.js (hunk 1 above)
Comment (boyloe) [reply src/Components/SurveyComponent.js]: That is right. I was thinking about it as if we wanted to call each individual question but we just need the survey data object. Good thinking, thanks.

**Review by eemshi — APPROVED** [review]
Looks good!

Screen Shot 2020-12-10 at 3 10 16 PM

**Pull request merged**